CavalcanteLucas/todolist
# OL-38: Login

Link to task:
https://labcodes.atlassian.net/browse/OL-38

How to test:
With a user created, fill the form for logging in at /;
Check if failure message appears when filling the “username” or “password” fields with invalid combinations of “username/password” or user is not registered;
If valid credentials are provided, check if redirection to /dashboard is occurring;
When logged in, a user menu interface should be visible;
When logged in, check if “Logout” button in the user menu is redirecting to /.


Description of your solution:
With this PR one can login with the registered credentials. This PR enables a form to logging in and a way to logging out of the App using Token-based authentication via API.
Login is considered successful if user can have access to her/his private dashboard of ToDo activities. Logout is considered successful the user Token is removed from the React store and a new user can now log in.
